crates/iroha_wasm_builder/src/lib.rs

```diff
@@ -181,17 +186,51 @@ mod internal {
 
         pub fn build(self) -> Result<Output> {
             let absolute_path = self.absolute_path.clone();
-            self.build_smartcontract().wrap_err_with(|| {
+            let optimize = self.profile == OPTIMIZED_PROFILE;
+            let output = self.build_smartcontract().wrap_err_with(|| {
                 format!(
                     "Failed to build the smartcontract at path: {}",
                     absolute_path.display()
                 )
-            })
+            })?;
+
+            if optimize {
+                let sp = if std::env::var("CI").is_err() {
+                    Some(spinoff::Spinner::new_with_stream(
+                        spinoff::spinners::Binary,
+                        "Optimizing the output",
+                        None,
+                        spinoff::Streams::Stderr,
+                    ))
+                } else {
+                    None
+                };
```

Comment:
I haven't thought about it. There is an option to set output printing using `builder.show_output()`. I don't see other uses for it than reporting errors. I included the printing spinner and confirmation message in lib mode when this flag is set.

Reply:
Can't the spinners be kept in `main.rs`?

Reply:
Since the optimization is deducted from the profile, both `cargo build` and `wasm-opt` are used in the `build()`. 
On the other hand, the spinner should be off when building but on when optimizing (`cargo build` -> `turn on spinner` -> `optimize`). I have tried to remove the spinner, yet new problems arose: the build logic repeats in `main.rs` and `lib.rs`, and some lib methods are forced to be public.

I hope there is a better approach.